testdata/rust/rust-1.62.0/plan.json

```diff
@@ -0,0 +1,24 @@
+{
+  "install_stage": {
+    "command": "",
+    "input_files": [
+      "."
+    ]
+  },
+  "build_stage": {
+    "command": ""
+  },
+  "start_stage": {
+    "command": "",
+    "input_files": [
+      "."
+    ]
+  },
+  "nix_overlays": [
+    "https://github.com/oxalica/rust-overlay/archive/master.tar.gz"
```

Comment:
Would it be better to pin to a specific version of the overlay, instead of doing master? Or because the overlay provides such granular control on which version of rust to install, it's actually good to just point at master (so we get any latest version of rust automatically)

Reply:
This is a good question. I spent some time reading the docs. Here's my mental model:

- The `oxalica/rust-overlay` updates daily the Rust-toolchain binaries it references..
- The specific code of the `oxalica/rust-overlay` evolves with each commit, so we should _not_ use `master` but instead pull from the `stable` tag's commit. This would be a soft-pin.
- I don't think we want to do a hard-pin to a specific commit, because projects which don't specify a Rust version would expect to get the newest stable Rust version. By pulling from `oxalica/rust-overlay:stable` tag, that code will get updated over time to point to newer versions of Rust.

In summary, I'll update this to use `https://github.com/oxalica/rust-overlay/archive/stable.tar.gz`

Reply:
One more bullet point:
- for projects that want a specific version of Rust, they include `rust-version` in their `Cargo.toml`. And we continue to respect that in the form of the actual nix-package that is pulled in.